plane/src/drone/executor.rs

```diff
@@ -53,13 +61,77 @@ impl<R: Runtime> Executor<R> {
 
         Self {
             runtime,
-            state_store: Arc::new(Mutex::new(state_store)),
+            state_store,
             backends,
             ip,
             _backend_event_listener: backend_event_listener,
         }
     }
 
+    // On restart, we want to terminate all existing backends and start fresh.
+    // This prevents bugs where an agent restart leaves the drone unable to
+    // terminate old backends.
+    async fn terminate_preexisting_backends(
+        runtime: Arc<R>,
+        state_store: Arc<Mutex<StateStore>>,
+    ) -> Result<()> {
+        let backends = state_store
+            .lock()
+            .expect("State store lock poisoned.")
+            .active_backends()?;
+
+        tracing::info!(?backends, "Terminating preexisting backends");
+
+        let mut tasks = vec![];
+        let state_store_clone = state_store.clone();
+        for (backend_id, state) in backends {
+            let runtime_clone = runtime.clone();
```

Comment:
Nit: elsewhere when we `clone()` to avoid ownership bugs, we often just use the same var name. Doing so doesn't change the behavior here, does it?

Reply:
It doesn't change the behavior; it just shadows that name within the block. I like that approach because it prevents you from referring to the wrong clone by mistake (since you literally can't refer to it later in the block) and can move code in/out of the block without renaming the var.

Reply:
Yep sorry just missed this one when switching styles, will fix